qa/pull-tester/rpc-tests.py

```diff
@@ -266,7 +268,16 @@ def main():
         subprocess.check_call((config["environment"]["SRCDIR"] + '/qa/rpc-tests/' + test_list[0]).split() + ['-h'])
         sys.exit(0)
 
-    run_tests(test_list, config["environment"]["SRCDIR"], config["environment"]["BUILDDIR"], config["environment"]["EXEEXT"], args.jobs, args.coverage, passon_args)
+    if (args.rpcgroup == -1) != (args.machines == -1):
+        print("ERROR: Please use both -m and -r options when using parallel rpc_groups.")
+        sys.exit(0)
+    if args.rpcgroup != -1 and args.machines != -1 and args.rpcgroup < args.machines:
+        k, m = divmod(len(test_list), args.machines)
+        k = k if m == 0 else k+1
+        split_list = list(test_list[i*k:min((i+1)*k, len(test_list))] for i in range(args.machines))
```

Comment:
The `min` here is unnecessary; per [note 4 of Common Sequence Operations](https://docs.python.org/3/library/stdtypes.html#common-sequence-operations), if the end of a slice is larger than the length, the length is substituted.
```suggestion
        split_list = list(test_list[i*k:(i+1)*k] for i in range(args.machines))
```

This could also be simplified further by using the `step` argument to `range`:
```suggestion
        split_list = list(test_list[i:i+k] for i in range(0, args.machines, k))
```